Components/Sources/Components/Utility/HTMLUtils.swift

```diff
@@ -0,0 +1,513 @@
+                continue
+            }
+            
+            attributedString.removeSubrange(tagRange)
+        }
+    }
+    
+    // MARK: - Shared - Private
+    
+    private static func htmlTagRegex() throws -> NSRegularExpression {
+        return try NSRegularExpression(pattern: "(?:<)([\\/a-z0-9]*)(?:\\s?)([^>]*)(?:>)")
+    }
+    
+    private static func subscriptPointSize(styles: Styles) -> CGFloat {
+        return styles.font.pointSize * 0.75
+    }
+    
+    private static func subscriptOffset(styles: Styles) -> CGFloat {
+        return -(styles.font.pointSize * 0.15)
+    }
+    
+    private static func superscriptPointSize(styles: Styles) -> CGFloat {
+        return styles.font.pointSize * 0.75
+    }
+    
+    private static func superscriptOffset(styles: Styles) -> CGFloat {
+        return styles.font.pointSize * 0.35
+    }
+    
+    private static func listInsertData(html: String) throws -> [ListInsertData] {
+        
+        // First enumerate through all list tags and capture insert data
+        let listTagRegex = try NSRegularExpression(pattern: "(?:<)((?:\\/?)(?:ol|ul|li))(?:\\s?)([^>]*)(?:>)")
+        var types: [ListType] = []
+        var indent: Int = 0
+        var orderedListCounts: [Int] = []
+        var inserts: [ListInsertData] = []
+        listTagRegex.enumerateMatches(in: html, range: html.fullNSRange) { match, flags, stop in
+            
+            guard let tagNSRange = match?.range(at: 0),
+                  let tagRange = Range(tagNSRange, in: html),
+                  let tagNameNSRange = match?.range(at: 1),
+                  let tagNameRange = Range(tagNameNSRange, in: html) else {
+                return
+            }
+            
+            let tagNameString = html[tagNameRange]
+            if tagNameString == "ol" {
+                types.append(.ordered)
+                indent = indent + 1
+                orderedListCounts.append(0)
+            } else if tagNameString == "ul" {
+                types.append(.unordered)
+                indent = indent + 1
+            }
+            
+            if tagNameString == "/ol" {
+                types.removeLast()
+                indent = indent - 1
+                orderedListCounts.removeLast()
+            } else if tagNameString == "/ul" {
+                types.removeLast()
+                indent = indent - 1
+            }
+            
+            if tagNameString == "li" {
+                
+                guard let currentListType = types.last else {
+                    return
+                }
+                
+                if case .ordered = currentListType {
+                    if let lastOrderedListCount = orderedListCounts.popLast() {
+                        orderedListCounts.append(lastOrderedListCount + 1)
+                    }
+                }
+                
+                let spaces = String(repeating: "    ", count: indent)
```

Comment:
Same note as `lineSpacing` above – I wonder if it makes sense to future-proof this with an indentation style element in `Styles` (either a `String` or `enum` to represent different indentation styles), and a static property on `HTMLUtils` that represents this string of spaces as a default.